103-signup-pages/signup-pages/src/js/main.js

```diff
@@ -0,0 +1,206 @@
+      nextPicture();
+    }
+  });
+
+  //读取文字数据
+  var dataPath = './data/picture/' + pid + '.json';
+  $.getJSON(dataPath, function(data){
+    var wordCount = data[0]['count'];
+    var wordData = data[0]['data'];
+
+    //随机图片中的文字，作为需要用户点击的文字
+    var wordId = Math.floor(Math.random() * 100) % wordCount + 1;
+    var wordValue = wordData[wordId].value;
+    pictureX = wordData[wordId].x;
+    pictureY = wordData[wordId].y;
+    $('#boxhint').text('请点击图片中的 ' + wordValue + ' 字');
+  });
+};
+
+//显示图片
+var showPicture = function(){
+  $('#boxhint').css('margin-left', '3%')
+      .css('color', 'white');
+  $box.html('<img src="./dist/images/no.png" />');
+  getPictureAndWord();
+  $picture.show();
+};
+
+//还原拖动Box
+var boxinit = function(){
+  $(document).unbind("mousemove");
+  $box.animate({
+    left: '0px'
+  });
+  $('.slide-show').animate({
+    width: '0px'
+  });
+};
+
+//Box拖动
+var moveSlideBox = function(){
+  //鼠标在box内按下触发
+  $box.mousedown(function (e){
+    var x = e.clientX - this.offsetLeft;
+    //这里mousemove与mouseup均注册到document，适应鼠标快速移动
+    $(document).mousemove(function(e){
+      var left = e.clientX - x;
+      if(left < 0){
+        left = 0;
+      }
+      //拖动到底后，去除拖动相关，显示图片
+      if(left > 272){
+        left = 272;
+        $(document).unbind("mousemove").unbind("mouseup");
+        $box.unbind("mousedown");
+        showPicture();
+      }
+      $box.css('left', left + 'px');
+      $('.slide-show').css('width', left + 'px');
+    });
+  });
+
+  $(document).mouseup(function(){
+    //鼠标释放，还原
+    boxinit();
+  });
+};
+
+var checkPhoneNumber = function(number){
+  if(number == ''){
+    $('#numberHint').show().text('请输入您的手机号码');
+    return;
+  }
+  var phoneReg = /^1[358]\d{9}$/;
+  if(phoneReg.test(number)){
+    $('#numberHint').hide();
+    return 1;
+  }else{
+    $('#numberHint').show().text('手机号码格式不正确, 请重新输入');
+    return 0;
+  }
+};
+
+var checkEmail = function(email){
+  if(email == ''){
+    $('#emailHint').show().text('请输入您的邮箱');
+    return;
+  }
+  var emailReg = /^[._\-a-z0-9]+@[._\-a-z0-9]+(\.[._\-a-z0-9]+)+$/;
+  if(emailReg.test(email)){
+    $('#emailHint').hide();
+    return 1;
+  }else{
+    $('#emailHint').show().text('邮箱格式不正确, 请重新输入');
+    return 0;
+  }
+};
+
+$(document).ready(function(){
+  moveSlideBox();
+
+  $('#phoneNumber').on('blur', function(){
+    var number = $('#phoneNumber').val();
+    checkPhoneNumber(number);
+  });
+
+  $('#submit').on('click', function(){
+    if(flag){
+      if(method == 1){
+        var number = $('#phoneNumber').val();
+        if(checkPhoneNumber(number) && flag){
+          $('#numberHint').hide();
+          alert('第一步成功!');
+        }
+      }
+      if(method == 2){
+        var email = $('#email').val();
+        if(checkEmail(email) && flag){
+          $('#numberHint').hide();
+          alert('第一步成功!');
+        }
+      }
+    }
+  });
+
+  //验证方式的切换
+  $('#regMethod').on('click', function(){
+    if(method == 1){
```

Comment:
should use something more meaningful than 1, like 'useEmail' or 'usePhoneNumber'